src/dotnet/Core/Commands/Internal/LocalCommandScheduler.cs

```diff
@@ -2,19 +2,65 @@ namespace ActualChat.Commands.Internal;
 
 public class LocalCommandScheduler : WorkerBase
 {
-    private LocalCommandQueue LocalCommandQueue { get; }
+    private readonly string _queueName;
+    private readonly int _degreeOfParallelism;
+    private readonly Action<string> _dispose;
+
+    private ICommandQueues CommandQueues { get; }
     private ICommander Commander { get; }
+    private EventCommander EventCommander { get; }
+    private ILogger<LocalCommandScheduler> Log { get; }
 
-    public LocalCommandScheduler(LocalCommandQueue localCommandQueue, ICommander commander)
+    public LocalCommandScheduler(
+        string queueName,
+        int degreeOfParallelism,
+        IServiceProvider serviceProvider,
+        Action<string> dispose)
     {
-        LocalCommandQueue = localCommandQueue;
-        Commander = commander;
+        _queueName = queueName;
+        _degreeOfParallelism = degreeOfParallelism;
+        _dispose = dispose;
+        CommandQueues = serviceProvider.GetRequiredService<ICommandQueues>();
+        Commander = serviceProvider.GetRequiredService<ICommander>();
+        EventCommander = serviceProvider.GetRequiredService<EventCommander>();
+        Log = serviceProvider.GetRequiredService<ILogger<LocalCommandScheduler>>();
     }
 
     protected override async Task RunInternal(CancellationToken cancellationToken)
     {
-        var commands = LocalCommandQueue.Commands.Reader.ReadAllAsync(cancellationToken);
-        await foreach (var command in commands.ConfigureAwait(false))
-            _ = Commander.Start(command, true, cancellationToken);
+        var queueReader = CommandQueues.Reader(_queueName, "");
+        var queuedCommands = queueReader.Read(cancellationToken);
+        await Parallel.ForEachAsync(
+            queuedCommands,
+            new ParallelOptions {
+                MaxDegreeOfParallelism = _degreeOfParallelism,
+                CancellationToken = cancellationToken,
+            },
+            async (queuedCommand, cancellationToken1) => {
+                try {
+                    if (queuedCommand.Command is IEvent)
+                        await EventCommander.Run(queuedCommand.Command, true, cancellationToken1).ConfigureAwait(false);
+                    else
+                        await Commander.Run(queuedCommand.Command, true, cancellationToken1).ConfigureAwait(false);
+                    await queueReader.Ack(queuedCommand, cancellationToken1).ConfigureAwait(false);
+                }
+                catch (OperationCanceledException e)
+                {
+                    await queueReader.NAck(queuedCommand, true, e, cancellationToken1).ConfigureAwait(false);
```

Comment:
If `cancellationToken1.IsCancellationRequested == true`, I suppose that `queueReader.NAck(queuedCommand, true, e, cancellationToken1)` will likely fail with OperationCanceledException as well. Am I wrong?